perfdash/metrics-downloader.go

```diff
@@ -279,6 +285,109 @@ func (b *GCSMetricsBucket) GetFile(job string, buildNumber int, path string) ([]
 	return data, nil
 }
 
+// S3MetricsBucket that creates a AWS S3 client to fetch data.
+type S3MetricsBucket struct {
+	client  *s3.S3
+	bucket  *string
+	logPath string
+}
+
+// NewS3MetricsBucket creates a new S3MetricsBucket.
+func NewS3MetricsBucket(bucket, path, region string) (MetricsBucket, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	s3Client := s3.New(sess)
+
+	return &S3MetricsBucket{
+		client:  s3Client,
+		bucket:  aws.String(bucket),
+		logPath: path,
+	}, nil
+}
+
+// GetBuildNumbers fetches the build numbers from an S3 Bucket.
+func (fetcher *S3MetricsBucket) GetBuildNumbers(job string) ([]int, error) {
```

Comment:
consider naming fetcher "b" or "s". It's typical go style for pointer receivers to be a single letter for brevity.